projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -542,17 +542,23 @@ export class IgxDropDownComponent extends IgxDropDownBaseDirective implements ID
         this.onSelection.emit(args);
 
         if (!args.cancel) {
-            this.selection.set(this.id, new Set([newSelection]));
-            if (!this.virtDir) {
-                if (oldSelection) {
-                    oldSelection.selected = false;
+            if (args.newSelection === null ||
+                (args.newSelection.hasOwnProperty('value') && args.newSelection.hasOwnProperty('index') && this.virtDir) ||
```

Comment:
Please, put that validation in a separate method with an appropriate name. Thanks!